---
title: "bash_rapture_pipeline"
author: "RP"
date: "11/9/2020"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

## Bash Rapture Pipeline

This is a basic overview of the pipeline for moving from raw seq data to bam files for analysis via `angsd`. Connect via `sftp farmer` to pull original files, then use git for remaining workflow.

See `/group/millermrgrp2/ryan/SEQS` for 2020 sequencing.

To use, `git clone` to the location, and make sure to make all code is executable: 

 - `chmod g+x *.pl`
 - `chmod g+x *.sh`

## Scripts

Scripts live in the `scripts` folder. This is an overview and notes about each script.

###  `01_unzip.sh`

This symlinks raw data from miller and copies it to the associated SOMM folder in SEQS. It then unzips the data for splitting.

 > if files already split: use in a screen: `for z in *fastq.gz; do gunzip -k $z; done`

 - **Run with:** `sbatch --mem MaxMemPerNode 01_unzip.sh SOMM504`

### `02_split_RAD.sh`

This is based on the original UCD Genome Center sequencing, and takes a set of barcodes of interest (based on unzipped `.fq` data from step 1) and splits by plate and then by sample using the barcodes provided.

This script requires the SOMM data directory, and depends on `BarcodeSplitList3Files.pl` and `BCsplitBestRadPE2.pl` perl scripts.

 - **Run with:** `sbatch --mem MaxMemPerNode 02_split_RAD.sh SOMM504`

### `03_split_wells.sh`

This is based on new (2020 sequencing data) and splits unzipped raw `.fq` data into individual folders for each plate, moves the `R1` and `R2` files into the plate folder, and then splits each plate into each well (sample).

This script requires a SOMM data directory and text file listing each plate barcode as an individual line, and depends on the `BarcodeSplit_RAD_PE.2019.pl` script (in `radseq_scripts/scripts`).

 - **Run with:** `sbatch -t 1200 -p high radseq_scripts/scripts/03_split_wells.sh SOMM504 plates.txt`

### `04_run_align.sh`

This aligns data using `samtools` and requires an indexed alignment file, as well as a list with the RA, RB, and outfile_name in 3 columns.

This script requires `bwa` and `samtools`, and can be run from wherever as long as the list provides the correct path to the fastq files (and associated output filepaths).

-- 

**TO MAKE ALIGN LIST**:

1. `ls *RA* | sed "s/\.fastq//g" > list_A`
2. `ls *RB* | sed "s/\.fastq//g" > list_B`
3. `ls *RA* | sed "s/\_RA//g" > list_noAB`
4. then paste together: `paste list_* > align_list`

**run altogether**
 - `ls *RA* > list_A; ls *RB* > list_B; ls *RA* | sed "s/\_RA//g" > list_noAB; paste list_* > align_list`

 - **Run with:** `sbatch --mem MaxMemPerNode 04_align.sh 3_col.bamlist reference_alignment.fa`